src/onnx/onnx.cpp

```diff
@@ -1430,6 +1435,24 @@ struct onnx_parser
         MIGRAPHX_THROW("Invalid tensor type");
     }
 
+    static literal
+    create_literal(shape::type_t shape_type, std::vector<size_t> dims, const char* data)
+    {
+        // in case of scalar constants in onnx file, use dims=1 to fill initializer data
+        if(dims.empty())
+            return literal{{shape_type, {1}, {0}}, data};
+        return literal{{shape_type, dims}, data};
+    }
+
+    template <class Iterator>
+    static literal
+    create_literal(shape::type_t shape_type, std::vector<size_t> dims, Iterator start, Iterator end)
```

Comment:
How about taking the container/range directly and then just calling `literal{{shape_type, dims}, data.begin(), data.end()}`? 

There is an issue with overloading here. You can change the name of the function, or use `MIGRAPHX_REQUIRES` to constraint it so it wont pick the wrong overload. You can probably write something like `template<class T, MIGRAPHX_REQUIRES(not std::is_pointer<T>{})>` to ensure that it wont be called for pointers.